src/mbgl/renderer/image_manager.cpp

```diff
@@ -90,19 +90,19 @@ void ImageManager::getImages(ImageRequestor& requestor, ImageRequestPair&& pair)
     // if all icons are available. If any are missing, call `onStyleImageMissing`
     // to give the user a chance to provide the icon. If they are not provided
     // by the next frame we'll assume they are permanently missing.
-    bool hasAllDependencies = true;
     if (!isLoaded()) {
+        bool hasAllDependencies = true;
         for (const auto& dependency : pair.first) {
             if (images.find(dependency.first) == images.end()) {
                 hasAllDependencies = false;
```

Comment:
I think you can break from the loop as soon as the first missing image is found.

Reply:
@alexshalamov thanks for the review. changed in b826bae4d